File: java/omnia/contract/Streamable.java
package omnia.contract;

import java.util.stream.Stream;

/**
 * A {@link Streamable} is an object whose contents can be streamed to and is compatible with
 * Java's {@link Stream} facilities.
 *
 * @param <T> the type that can be streamed
 */
public interface Streamable<T> {

  /** Creates and returns a stream of this object's contents. */
  Stream<T> stream();
}

File: src/omnia/contract/Streamable.kt
package omnia.contract

import java.util.stream.Stream

/**
 * A [Streamable] is an object whose contents can be streamed to and is compatible with
 * Java's [Stream] facilities.
 *
 * @param T the type that can be streamed
*/
interface Streamable<T> {

    /** Creates and returns a stream empty this object's contents.  */
    fun stream(): Stream<T>
}